base/errorshow.jl

```diff
@@ -518,18 +517,12 @@ function show_reduced_backtrace(io::IO, t::Vector, with_prefix::Bool)
         end
 
         if repetitions==0
-            push!(displayed_stackframes, (last_frame, n))
-        end
-    end
-
-    try invokelatest(update_stackframes_callback[], displayed_stackframes) catch end
-
-    for (frame, n) in displayed_stackframes
-        if with_prefix
-            show_trace_entry(io, frame, n, prefix = string(" [", frame_counter-1, "] "))
-            push!(LAST_SHOWN_LINE_INFOS, (string(frame.file), frame.line))
-        else
-            show_trace_entry(io, frame, n)
+            if with_prefix
+                show_trace_entry(io, last_frame, n, prefix = string(" [", frame_counter-1, "] "))
+                push!(LAST_SHOWN_LINE_INFOS, (string(last_frame.file), last_frame.line))
```

Comment:
I think you need to capture the corrected frame. Otherwise `LAST_SHOWN_LINE_INFOS` will be the "old" stacktrace, but what's displayed in the REPL will be the new one. Then when someone hits `15^Q` to go to the point in the file they will wonder why `edit` opens the file to the wrong line number.